pkg/cluster/internal/providers/podman/provider.go

```diff
@@ -183,19 +183,41 @@ func (p *Provider) GetAPIServerEndpoint(cluster string) (string, error) {
 	}
 
 	// portMapping maps to the standard CNI portmapping capability
-	// see: https://github.com/containernetworking/cni/blob/spec-v0.4.0/CONVENTIONS.md
-	type portMapping struct {
+	type portMapping19 struct {
 		HostPort      int32  `json:"hostPort"`
 		ContainerPort int32  `json:"containerPort"`
 		Protocol      string `json:"protocol"`
 		HostIP        string `json:"hostIP"`
 	}
+	type portMapping20 struct {
```

Comment:
can you link to some place explaining this new struct?
is podman 2.0 not using CNI?
portMapping20 seems a subset of portMapping19, is not possible to use the same struct?

Reply:
it is coming from: https://github.com/containers/podman/blob/master/libpod/define/container_inspect.go#L130-L139

@mheon should we consider it a breaking change?

They also have different types (one is int32 and the other is string)

Reply:
2.0 changed the output of inspect port mapping to match Docker, as opposed to previous versions which directly dumped the JSON form of CNI port mappings. Generally speaking, in the past, Podman has considered these bugfixes (we declare that our CLI is Docker compatible, so unintentional breakages like this are bugs).

Reply:
please add comments to the portMapping structs explaining its sources, something along these lines
```go
 	// portMapping20 maps to the podman 2.0 portmap type
	// see:  https://github.com/containers/podman/blob/master/libpod/define/container_inspect.go#L130-L139
```

Reply:
added

Reply:
> Generally speaking, in the past, Podman has considered these bugfixes (we declare that our CLI is Docker compatible, so unintentional breakages like this are bugs).

@mheon podman will have the same policy for 2.0 onwards?

Reply:
Yes, but I don't think we're expecting any further changes here. Inspect now matches Docker to a high degree, so we've attained our goal of compatibility.